Source/Core/Core/HW/EXI/EXIBrawlback.cpp

```diff
@@ -1189,137 +1191,328 @@ void CEXIBrawlback::handleStartMatch(u8* payload)
 
 void CEXIBrawlback::fixStartReplayEndianness(StartReplay& startReplay)
 {
-  swap_endian(startReplay.firstFrame);
-  swap_endian(startReplay.otherRandomSeed);
-  swap_endian(startReplay.randomSeed);
+  startReplay.firstFrame = swap_endian(startReplay.firstFrame);
+  startReplay.otherRandomSeed = swap_endian(startReplay.otherRandomSeed);
+  startReplay.randomSeed = swap_endian(startReplay.randomSeed);
   for (int i = 0; i < startReplay.numPlayers; i++)
   {
-    swap_endian(startReplay.players[i].startPlayer.xPos);
-    swap_endian(startReplay.players[i].startPlayer.yPos);
-    swap_endian(startReplay.players[i].startPlayer.zPos);
+    auto& startPlayer = startReplay.players[i].startPlayer;
+
+    startPlayer.xPos = swap_endian(startPlayer.xPos);
+    startPlayer.yPos = swap_endian(startPlayer.yPos);
+    startPlayer.zPos = swap_endian(startPlayer.zPos);
   }
 }
 
 void CEXIBrawlback::handleStartReplaysStruct(u8* payload)
 {
-  StartReplay startReplay;
-  std::memcpy(&startReplay, payload, sizeof(StartReplay));
-  fixStartReplayEndianness(startReplay);
```

Comment:
You removed fixStartReplayEndianess here, but it doesn't look like it's being called at all anymore. Is it still needed?